lib/hitobito_sww/wagon.rb

```diff
@@ -39,6 +39,13 @@ class Wagon < Rails::Engine
                                            :name_add_on, :title]
       InvoicesController.permitted_attrs += [:membership_card, :membership_expires_on]
 
+      # Since permitted_attrs are an array, it's really hard to expand nested attrs
+      # rubocop:disable Metrics/LineLength
+      invoice_lists_invoice_permitted_attrs_hash = InvoiceListsController.permitted_attrs.find { |attr| attr.is_a?(Hash) && attr.keys.include?(:invoice) }
```

Comment:
Ich glaube es wäre es wert, wenn du im Core in InvoiceListsController eine neue Klassenvariable oder [mutierbare Konstante](https://github.com/hitobito/hitobito/blob/master/app/models/person.rb#L67) einfügst, die dann in Wagons einfacher erweitert werden kann.
```ruby
self.permitted_attrs = [
  :receiver_id,
  :receiver_type,
  :recipient_ids,
  invoice: self.permitted_invoice_attrs, # untested, use at own risk
]
```
Also ich bin nicht zu 100% sicher ob das so dann funktioniert mit Überschreiben, aber wenn du eh gerade einen PR im Wagon und im Core offen hast könntest du es noch probieren. Es würde definitiv zu schönerem Code führen.

Reply:
Leider nicht möglich, da die `permitted_attrs` nicht rückwirkend "aktualisiert" werden wenn `permitted_invoice_attrs`
Grundsätzlich wäre sowas mit einer `permitted_attrs` Methode möglich. Das aber rückwirkend in den Controller einzubauen wäre wegen Auswirkungen auf allfällige Wagons etwas zu umständlich